terraform/lib/dependabot/terraform/file_updater.rb

```diff
@@ -18,14 +19,18 @@ def updated_dependency_files
         updated_files = []
 
         [*terraform_files, *terragrunt_files].each do |file|
-          next unless file_changed?(file)
+          next unless file_changed?(file) || lock_file?(file.name)
 
           updated_content = updated_terraform_file_content(file)
-          raise "Content didn't change!" if updated_content == file.content
+
+          raise "Content didn't change!" unless lockfile_changed? || updated_content != file.content
 
           updated_files << updated_file(file: file, content: updated_content)
         end
 
+        updated_files << update_lockfile_declaration(dependency) unless lock_file.empty?
```

Comment:
Small nit but might be clearer to not pass any arguments to this method as both `dependency` and `lock_file` are available in the class

Reply:
Removed the dependency argument and moved the `lock_file.nil?` check to inside the `update_lockfile_declaration` method.